File: Code Gladiators 2020/Open Coding Round/The Powerpuff Girls.cpp
#include <bits/stdc++.h>
using namespace std;

int main(int argc, char *a[]) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    cin >> N;

    long long requirements[N], quantity[N], ans = LLONG_MAX; 
       
    for(int i=0; i<N; i++) cin >> requirements[i]; 
    for(int i=0; i<N; i++) cin >> quantity[i]; 
        
    for(int i=0; i<N; i++) ans = min(ans, quantity[i]/requirements[i]); 

    cout << ans << endl;
}

File: Code Gladiators 2020/Open Coding Round/The Powerpuff Girls.rb
def main()
    n, ans = gets.chomp.to_i, 9223372036854775807
    
    requirements = gets.chomp.split.map(&:to_i)
    quantity = gets.chomp.split.map(&:to_i)
    
    (0..n-1).each { |i| ans = [ans, quantity[i]/requirements[i]].min }

    p ans
end
main()
